### DISCUSSION TITLE
Welcome to emp Discussions!

### DISCUSSION THREAD

**Discussion by ckken** [empjs/emp#52] [Ideas]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by flyyuan** [comment]
😆😆😆

**Comment by yuki070** [comment]
有几个场景不知道现实中EMP是怎么处理的：

- 如果我有4个项目，每个项目间的组件都有互相依赖的，那么我在开发其中一个项目的时候，都必须把其他3个项目跑起来？
- 如果项目A引用了项目B的组件`X`的最初版本，项目A上线后，非常稳定。但项目B对组件`X`进行了调整，而且带来了一些breakChange，那么项目B上线后，项目A是否就没法运行了？

**Reply by wengdongyang** [reply]
我做了如下处理。
1.base部分，分为base1，base2分别作为最底层的组件依赖（包括全局的layout，公共组件等）（这部分不要轻易改动）
2.project部分，分为project1，project2，它只依赖base部分，输出的其实是一个个的页面模块。页面之间不要有逻辑上或者其他耦合，也最好不要相互依赖。
3.export部分，这部分是最终的输出产物。他的依赖是所有的base和project。

**Reply by yuki070** [reply]
`不要轻易改动`这种是靠人为去遵循的，还是没有很好地解决依赖组件的不同版本的问题。
